tests/test_join.py

```diff
@@ -316,15 +316,6 @@ def test_table_multiple_self_join(db: gp.Database, zoo_1: gp.Table):
 def test_join_recursive_join(db: gp.Database):
     rows = [(i,) for i in range(10)]
     numbers = gp.values(rows, db=db, column_names=["val"])
-    label = numbers[["*", (numbers["val"] % 2).rename("mod")]]
-    percentile = label[["percentile_disc(0.5) WITHIN GROUP (ORDER BY mod) AS mod"]]
-    ret_mod = list(
-        label.inner_join(
-            percentile,
-            (label["mod"] == percentile["mod"]),
-            targets=[label["val"], percentile["mod"]],
-        ).fetch()
-    )
-    assert len(ret_mod) == 5
-    for row in ret_mod:
-        assert row["mod"] == row["val"] % 2
+    label = numbers.extend("mod", numbers["val"] % 2)
+    results = label.inner_join(numbers, numbers["val"] == label["val"], targets=[label["val"]])
```

Comment:
It seems like there has one layer less before join, this handle a specific recursive join test case.